### PR TITLE
Implement parameter binding for the query function

### PR DESCRIPTION
**Pull request by findolor**
## Motivation

See issue: https://github.com/rainlanguage/sqlite-web/issues/18

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

* **New Features**
  * Query API now accepts optional parameters end-to-end, with support for BigInt and binary/blob values and structured worker query results.

* **Bug Fixes / Reliability**
  * Improved parameter validation/binding, clearer parameter-related errors, single-statement enforcement for parameterized queries, and more robust worker-side error reporting and timeouts.

* **Tests**
  * Expanded integration and wasm tests covering positional/numbered params, types, edge cases, multi-statement and cross-worker routing.

* **Chores**
  * Test fixtures updated and JS interop helpers added for param normalization and value handling; package artifact bumped.

### CODE REVIEW

File: packages/sqlite-web-core/src/worker.rs [hunk 1]
```diff
@@ -42,59 +42,106 @@ pub fn main() -> Result<(), JsValue> {
...
+                                                        )
+                                                        .unwrap();
+                                                    }
+                                                };
+
+                                                let global = js_sys::global();
+                                                let worker_scope: DedicatedWorkerGlobalScope =
+                                                    global.unchecked_into();
+                                                let _ = worker_scope.post_message(&response);
+                                            });
```
[the remark this answers is not in the document]
Comment (findolor) [reply packages/sqlite-web-core/src/worker.rs]: not really needed

File: packages/sqlite-web/src/lib.rs [hunk 2]
```diff
@@ -178,32 +179,155 @@ impl SQLiteWasmDatabase {
...
+                    &obj,
+                    &JsValue::from_str("__type"),
+                    &JsValue::from_str("blob"),
+                )
+                .map_err(SQLiteWasmDatabaseError::from)?;
+                js_sys::Reflect::set(&obj, &JsValue::from_str("base64"), &JsValue::from_str(&b64))
+                    .map_err(SQLiteWasmDatabaseError::from)?;
+                normalized.push(&obj);
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (findolor) [reply packages/sqlite-web/src/lib.rs]: not needed at this moment

File: packages/sqlite-web-core/src/database.rs [hunk 3]
```diff
@@ -13,7 +15,420 @@ pub struct SQLiteDatabase {
...
+                        return false;
+                    }
+                }
+                _ => return false,
+            }
+        }
+        true
+    }
+
+    fn bind_params_for_stmt(
```
Comment (0xgleb) [inline packages/sqlite-web-core/src/database.rs]: this function is **seven** screens long. can we, please, refactor it? Also, I reckon a lot of the mutability and looping can be replaced with functional-style iterator method chaining, which could be easier to read/comprehend, so consider using that as well

File: packages/sqlite-web-core/src/worker.rs [hunk 4]
```diff
@@ -36,68 +37,83 @@ pub fn main() -> Result<(), JsValue> {
...
-                                            }
-                                        };
-
-                                        let global = js_sys::global();
-                                        let worker_scope: DedicatedWorkerGlobalScope =
-                                            global.unchecked_into();
-                                        let _ = worker_scope.post_message(&response);
-                                    });
+        // Deserialize into canonical WorkerMessage and handle variants
+        match serde_wasm_bindgen::from_value::<WorkerMessage>(data) {
```
Comment (0xgleb) [inline packages/sqlite-web-core/src/worker.rs]: i see multiple .unwrap's in here. this isn't test code, is it?

**Review by 0xgleb — CHANGES_REQUESTED** [review]

File: packages/sqlite-web-core/src/database_functions/mod.rs [hunk 5]
```diff
@@ -19,7 +19,8 @@ use float_zero_hex::*;
 /// Register all custom functions with the SQLite database
 pub fn register_custom_functions(db: *mut sqlite3) -> Result<(), String> {
     // Register BIGINT_SUM aggregate function
-    let bigint_sum_name = CString::new("BIGINT_SUM").unwrap();
+    let bigint_sum_name = CString::new("BIGINT_SUM")
+        .map_err(|_| "Function name BIGINT_SUM contains interior NUL bytes".to_string())?;
     let ret = unsafe {
```
[the remark this answers is not in the document]
Comment (findolor) [reply packages/sqlite-web-core/src/database_functions/mod.rs]: not needed

File: packages/sqlite-web-core/src/util.rs [hunk 6]
```diff
@@ -22,6 +26,32 @@ pub fn sanitize_db_filename(name: &str) -> String {
...
 
+pub fn set_js_property(target: &JsValue, key: &str, value: &JsValue) -> Result<(), JsValue> {
+    match Reflect::set(target, &JsValue::from_str(key), value) {
+        Ok(true) => Ok(()),
+        Ok(false) => Err(JsValue::from_str(&format!(
+            "Reflect::set returned false for key {key}"
+        ))),
+        Err(err) => Err(err),
+    }
+}
```
[the remark this answers is not in the document]
Comment (findolor) [reply packages/sqlite-web-core/src/util.rs]: not needed

File: packages/sqlite-web/src/lib.rs [hunk 7]
```diff
@@ -11,6 +12,150 @@ use web_sys::{Blob, BlobPropertyBag, MessageEvent, Url, Worker};
...
+    js_sys::Reflect::set(
+        &obj,
+        &JsValue::from_str("__type"),
+        &JsValue::from_str("blob"),
+    )
+    .map_err(SQLiteWasmDatabaseError::from)?;
+    js_sys::Reflect::set(&obj, &JsValue::from_str("base64"), &JsValue::from_str(&b64))
+        .map_err(SQLiteWasmDatabaseError::from)?;
+    Ok(obj.into())
+}
```
[the remark this answers is not in the document]
Comment (findolor) [reply packages/sqlite-web/src/lib.rs]: not needed

File: packages/sqlite-web-core/src/database.rs [hunk 8]
```diff
@@ -13,7 +15,565 @@ pub struct SQLiteDatabase {
...
+        let last = buffers._texts.last().unwrap();
+        let ptr = last.as_ptr();
+        let len = last.as_bytes().len() as i32;
+        let rc = unsafe {
+            sqlite3_bind_text(stmt, i, ptr, len, None::<unsafe extern "C" fn(*mut c_void)>)
+        };
+        if rc != SQLITE_OK {
+            return Err(format!("Failed to bind text at {i}"));
+        }
+        Ok(())
```
[the remark this answers is not in the document]
Comment (findolor) [reply packages/sqlite-web-core/src/database.rs]: not needed

File: packages/sqlite-web-core/src/worker.rs [hunk 9]
```diff
@@ -2,27 +2,134 @@
...
+                o
+            });
+            if let Err(err) = post_message(&response) {
+                if let Err(send_err) = send_worker_error(err) {
+                    throw_val(send_err);
+                }
+            }
+        }
+    }
+}
```
Comment (0xgleb) [inline packages/sqlite-web-core/src/worker.rs]:
```suggestion
fn handle_incoming_value(data: JsValue) {
    let make_error_response = || {
        let o = js_sys::Object::new();
        let _ = set_js_property(&o, "type", &JsValue::from_str("query-result"));
        let _ = set_js_property(&o, "result", &JsValue::NULL);
        let _ = set_js_property(&o, "error", &JsValue::from_str("Failed to build response"));
        o
    };
    
    let send_response = |response: &js_sys::Object| {
        post_message(response)
            .or_else(|err| send_worker_error(err).map_err(throw_val))
            .ok();
    };

    match serde_wasm_bindgen::from_value::<WorkerMessage>(data) {
        Ok(WorkerMessage::ExecuteQuery { sql, params }) => {
            WORKER_STATE.with(|s| {
                let Some(state) = s.borrow().as_ref().map(Rc::clone) else {
                    return;
                };
                spawn_local(async move {
                    let result = state.execute_query(sql, params).await;
                    let response = make_query_result_message(result).unwrap_or_else(|err| {
                        let _ = send_worker_error(err);
                        make_error_response()
                    });
                    send_response(&response);
                });
            });
        }
        Err(err) => {
            let error_text = format!("Invalid worker message: {err:?}");
            let response = make_query_result_message(Err(error_text)).unwrap_or_else(|e| {
                let _ = send_worker_error(e);
                make_error_response()
            });
            send_response(&response);
        }
    }
}
```
